controllers/policyset/policyset_controller.go

```diff
@@ -20,24 +33,217 @@ var log = ctrl.Log.WithName(ControllerName)
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		log.Error(err, "Failed to retrieve policy set")
+
+		return reconcile.Result{}, err
+	}
+	log.V(1).Info("Policy set was found, processing it...")
 
-	return ctrl.Result{}, nil
+	setNeedsUpdate := processPolicySet(r.Client, instance)
+
+	if setNeedsUpdate {
+		log.V(1).Info("Status update needed...")
+		faultyPlcSet, err := updatePolicySetStatus(r.Client, instance)
+		if err != nil {
+			log.Error(err, fmt.Sprintf("reason: policy update error: policy/%v, namespace: %v",
+				faultyPlcSet.Name, faultyPlcSet.Namespace))
+
+			return reconcile.Result{}, err
+		}
+	}
+	log.Info("Policy set successfully processed, reconcile complete")
+	return reconcile.Result{}, nil
+}
+
+// processPolicySet compares the status of a policyset to its desired state and determines whether an update is needed
+func processPolicySet(c client.Client, plcSet *policyv1.PolicySet) bool {
+	log.V(1).Info("Processing policy sets...")
+
+	needsUpdate := false
+
+	// compile results and compliance state from policy statuses
+	generatedResults := []policyv1.PolicySetStatusResult{}
+	complianceFound := false
+	aggregatedCompliance := "Compliant"
+	placementsByBinding := map[string]policyv1.PolicySetStatusPlacement{}
+
+	for i := range plcSet.Spec.Policies {
+		childPlcName := plcSet.Spec.Policies[i]
+		childNamespacedName := types.NamespacedName{
+			Name:      string(childPlcName),
+			Namespace: plcSet.Namespace,
+		}
+		childPlc := &policyv1.Policy{}
+		err := c.Get(context.TODO(), childNamespacedName, childPlc)
+		if err != nil {
+			// policy does not exist, return error message
+			var errMessage string
+			if errors.IsNotFound(err) {
+				errMessage = string(childPlcName) + " not found"
+			} else {
+				errMessage = strings.Split(err.Error(), "Policy.policy.open-cluster-management.io ")[1]
+			}
+			generatedResults = append(generatedResults, policyv1.PolicySetStatusResult{
+				Policy:  string(childPlcName),
+				Message: errMessage,
+			})
+		} else {
+			// policy exists - can use it to calculate status data
+			log.V(1).Info("Evaluating changes in policy " + string(childPlcName) + "...")
+			generatedResults = append(generatedResults, policyv1.PolicySetStatusResult{
+				Policy:    string(childPlcName),
+				Compliant: string(childPlc.Status.ComplianceState),
+				Clusters:  statusToClusters(childPlc.Status.Status),
+			})
+
+			// aggregate compliance state
+			if string(childPlc.Status.ComplianceState) != "" {
+				complianceFound = true
+			}
+			if string(childPlc.Status.ComplianceState) == "NonCompliant" {
+				aggregatedCompliance = "NonCompliant"
+			}
+
+			// aggregate placements
+			for _, placement := range childPlc.Status.Placement {
+				if placement.PolicySet == plcSet.GetName() {
+					placementsByBinding[placement.PlacementBinding] = plcPlacementToSetPlacement(*placement)
+				}
+			}
+		}
+	}
+	if !reflect.DeepEqual(generatedResults, plcSet.Status.Results) {
```

Comment:
Can you check if this works the same as `equality.Semantic`? If yes, let's use `equality.Semantic`

Reply:
Can you also explain why do you have to compare each sub field of status instead of compare the entire status object and then determine if update is needed?

Reply:
the code is set up to check each sub field so I could add logs to see which fields it flagged as needing an update. Those logs aren't very useful outside of debugging purposes so I will modify it to check the entire status.